### ISSUE TITLE
Header back button accessibilityLabel has hardcoded "back" and it cannot be changed from stack navigator

### ISSUE THREAD

**Issue by mskri** [react-navigation/react-navigation#9016]
**Current Behavior**

Setting `headerBackTitle` to a stack navigator announces back button on screen reader as "<title>, back, button". For proper i18n and accessibility of the back button this is problematic as the ", back" part is not properly translated. This is a problem when the app is not in English. Rest of the label can be in different language but the "back" will be in English.

If you look e.g. at Apple's settings app the back button does not have ", back" as part of the accessibility label it only uses the label of the back button.

When setting the left button of stack navigator manually you can use `<HeaderBackButton />` and circumvent this problem by setting the `accessibilityLabel` directly to what you want it to be. However if you are not setting the back button manually and rely on stack navigator to do it for you then you cannot remove the hardcoded "back" from the accessibility label.

**Expected Behavior**

- `HeaderBackButton` does not have hardcoded ", back" as part of the accessibility label
- You can set accessibility label for a header back button can be set from stack navigator

**How to reproduce**

Let's say you have an app in Finnish and you have set the `headerBackTitle` on a stack navigator to "Koti" (meaning "home" in English). You have a multiple screens on the navigator and you navigate to one. Stack navigator adds the header back button automatically for you to navigate back. Screen reader will read the accessibility label from that header back button as "Koti, back, painike" (painike = button in Finnish). The accessibility role "button" will be translated to "painike" based on the OS language.

**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 | ios |
| @react-navigation/native       | 5.8.2
| @react-navigation/stack        | 5.11.1
| react-native-gesture-handler   |
| react-native-safe-area-context |
| react-native-screens           |
| react-native                   | 0.63.2
| expo                           |
| node                           | 
| npm or yarn                    | npm
